src/ManifoldsBase.jl

```diff
@@ -64,37 +64,52 @@ representations of cotangent vectors and their types on a manifold.
 abstract type CoTVector end
 
 """
-    base_manifold(M::Manifold)
+    is_decorator_manifold(M)
+
+indicate whether a manifold is a decorator manifold, i.e. whether it encapsulates
+a [`Manifold`](@ref) with additional features, and stores internally the original
+manifold instance. An example is the [`ArrayManifold`](@ref).
+
+Using Tim Holys Traits Trick (THTT), certain functions are just calling themselves
+on the internal manifold, and hence have not to be implemented for decorators
+again, for example [`manifold_dimension`](@ref) and especially [`base_manifold`](@ref).
+This also assumes, that the undecoratord (base) manifold is stored in `M.manifold`.
```

Comment:
Perhaps it could be 'decorated' instead of 'undecoratord'?

Reply:
Or one could leave that just out, because within might be a decorated or a non decorated manifold; we might encapsulate quite a few decorators.

Reply:
Yes, that's also an option. By 'decorated' I meant earlier the encapsulated manifold but it can be somewhat confusing.